### PR TITLE
Dpr 445 additional schemas

### PR DESCRIPTION
**Pull request by kartthik-n**

### CODE REVIEW

File: src/test/resources/contracts/agency-internal-locations.avsc [hunk 1]
```diff
@@ -9,148 +15,179 @@
...
-      "description": "The unique identifier of the Internal Location. PK",
-      "glossary-index": "location:id",
       "name": "INTERNAL_LOCATION_ID",
+      "type": "long",
       "nullable": false,
-      "key": "primary",
-      "sensitivity": "public",
-      "type": "long"
+      "description": "The unique identifier of the Internal Location. PK",
+      "glossary-index": "location:id",
```
Comment (dominic-messenger-moj) [inline src/test/resources/contracts/agency-internal-locations.avsc:24]: Should be internal-location:id

File: src/test/resources/contracts/agency-locations.avsc [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+      "name": "DESCRIPTION",
+      "type": "string",
+      "nullable": true,
+      "description": "Description of the agency location."
+    },
+    {
+      "name": "AGENCY_LOCATION_TYPE",
+      "type": "string",
+      "nullable": true,
+      "glossary-index": "location:id",
```
Comment (dominic-messenger-moj) [inline src/test/resources/contracts/agency-locations.avsc:37]: Pls remove glossary-index for AGENCY_LOCATION_TYPE

File: src/test/resources/contracts/agency-locations.avsc [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+      "description": "Reference Code [ AGY_LOC_TYPE ]",
+      "key": "foreign"
```
Comment (dominic-messenger-moj) [inline src/test/resources/contracts/agency-locations.avsc:39]: This is a reference code, which is probably represented as an Enum. Needs discussion with BA on how reference codes will be represented.
Not a foreign key

File: src/test/resources/contracts/agency-locations.avsc [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+      "name": "CONTACT_NAME",
+      "type": "string",
+      "nullable": true,
+      "description": "Contact person at agency location."
+    },
+    {
+      "name": "PRINT_QUEUE",
+      "type": "string",
+      "nullable": true,
+      "glossary-index": "location:id",
```
Comment (dominic-messenger-moj) [inline src/test/resources/contracts/agency-locations.avsc:78]: Remove glossaryindex

File: src/test/resources/contracts/offender-bookings.avsc [hunk 5]
```diff
@@ -0,0 +1,358 @@
...
+      "nullable": true,
+      "description": "The date the booking was created."
+    },
+    {
+      "name": "BOOKING_END_DATE",
+      "type": {
+        "type": "int",
+        "logicalType": "date"
+      },
+      "nullable": false,
```
Comment (dominic-messenger-moj) [inline src/test/resources/contracts/offender-bookings.avsc:42]: nullable is true

File: src/test/resources/contracts/offender-bookings.avsc [hunk 6]
```diff
@@ -0,0 +1,358 @@
...
+      "glossary-index": "offender_bookings:id",
+      "key": "primary"
+    },
+    {
+      "name": "BOOKING_BEGIN_DATE",
+      "type": {
+        "type": "int",
+        "logicalType": "date"
+      },
+      "nullable": true,
```
Comment (dominic-messenger-moj) [inline src/test/resources/contracts/offender-bookings.avsc:33]: nullable is false

File: src/test/resources/contracts/offender-bookings.avsc [hunk 7]
```diff
@@ -0,0 +1,358 @@
...
+      "type": "string",
+      "nullable": true,
+      "glossary-index": "location:id",
+      "description": "The location residing within an agency."
+    },
+    {
+      "name": "LIVING_UNIT_ID",
+      "type": "long",
+      "nullable": true,
+      "glossary-index": "living_unit:id",
```
Comment (dominic-messenger-moj) [inline src/test/resources/contracts/offender-bookings.avsc:69]: Isn't this the internal-locations:id ? Should the internal-locations:id actually be called living-unit:id ?  
Please clarify with Michael/Zahoor and ensure this schema and internal-locations schema are the same.

File: src/test/resources/contracts/offender-bookings.avsc [hunk 8]
```diff
@@ -0,0 +1,358 @@
...
+    {
+      "name": "BOOKING_STATUS",
+      "type": "string",
+      "nullable": true,
+      "description": "Reference Code ( BOOK_STS ). The open or closed status of the booking."
+    },
+    {
+      "name": "YOUTH_ADULT_CODE",
+      "type": "string",
+      "nullable": true,
```
Comment (dominic-messenger-moj) [inline src/test/resources/contracts/offender-bookings.avsc:99]: nullable is false

File: src/test/resources/contracts/offender-bookings.avsc [hunk 9]
```diff
@@ -0,0 +1,358 @@
...
+    {
+      "name": "FINGER_PRINTED_STAFF_ID",
+      "type": "long",
+      "nullable": true,
+      "description": "The staff member finger printing the offender."
+    },
+    {
+      "name": "SEARCH_STAFF_ID",
+      "type": "long",
+      "nullable": false,
```
Comment (dominic-messenger-moj) [inline src/test/resources/contracts/offender-bookings.avsc:111]: nullable is true

File: src/test/resources/contracts/offender-bookings.avsc [hunk 10]
```diff
@@ -0,0 +1,358 @@
...
+      "description": "Id of staff member who searched offender."
+    },
+    {
+      "name": "PHOTO_TAKING_STAFF_ID",
+      "type": "long",
+      "nullable": false,
```
Comment (dominic-messenger-moj) [inline src/test/resources/contracts/offender-bookings.avsc:117]: nullable is true

File: src/test/resources/contracts/offender-bookings.avsc [hunk 11]
```diff
@@ -0,0 +1,358 @@
...
+      "type": "long",
+      "nullable": true,
+      "glossary-index": "offender:id",
+      "description": "The Root Offender Identifier"
+    },
+    {
+      "name": "AGENCY_IML_ID",
+      "type": "long",
+      "nullable": true
+    },
```
Comment (dominic-messenger-moj) [inline src/test/resources/contracts/offender-bookings.avsc:155]: should be a glossary index foreign key to internal-locations:id or living-unit_id (whichever is decided)

File: src/test/resources/contracts/offender-bookings.avsc [hunk 12]
```diff
@@ -0,0 +1,358 @@
...
+    {
+      "name": "STATUS_REASON",
+      "type": "string",
+      "nullable": true,
+      "description": "For Ontario Header Status"
+    },
+    {
+      "name": "TOTAL_UNEXCUSED_ABSENCES",
+      "type": "long",
+      "nullable": false,
```
Comment (dominic-messenger-moj) [inline src/test/resources/contracts/offender-bookings.avsc:257]: Nullable is true

File: src/test/resources/contracts/offender-bookings.avsc [hunk 13]
```diff
@@ -0,0 +1,358 @@
...
+    },
+    {
+      "name": "AUDIT_ADDITIONAL_INFO",
+      "type": "string",
+      "nullable": true
+    },
+    {
+      "name": "BOOKING_SEQ",
+      "type": "long",
+      "nullable": true
```
Comment (dominic-messenger-moj) [inline src/test/resources/contracts/offender-bookings.avsc:350]: nullable is false

**Review by dominic-messenger-moj — CHANGES_REQUESTED** [review]
Many little errors.
Please review the key for agency-internal-locations
Should be living-unit:id for primary key or internal-location:id - depending on what Michael and Zahoor agree.

File: src/test/resources/contracts/offender-bookings.avsc (hunk 10 above)
Comment (kartthik-n) [reply src/test/resources/contracts/offender-bookings.avsc:117]: done

File: src/test/resources/contracts/offender-bookings.avsc (hunk 11 above)
Comment (kartthik-n) [reply src/test/resources/contracts/offender-bookings.avsc:155]: checked with Zahoor and he confirmed glossary-index is not part of this ticket and can be handled in another backlog ticket he is working on

File: src/test/resources/contracts/agency-locations.avsc (hunk 3 above)
Comment (kartthik-n) [reply src/test/resources/contracts/agency-locations.avsc]: Enum is not part of this ticket and all reference data not part of SOW1. confirmed by Zahoor

File: src/test/resources/contracts/offender-bookings.avsc (hunk 6 above)
Comment (dominic-messenger-moj) [reply src/test/resources/contracts/offender-bookings.avsc:33]: Will this be resolved ?

File: src/test/resources/contracts/offender-bookings.avsc (hunk 7 above)
Comment (dominic-messenger-moj) [reply src/test/resources/contracts/offender-bookings.avsc:69]: It's called internal-location:id in the agency_internal_locations. This needs to be the SAME.

**Review by dominic-messenger-moj — CHANGES_REQUESTED** [review]
Still some inconsistencies.

**Review by dominic-messenger-moj — APPROVED** [review]

**Pull request merged**